src/features/defi/providers/thorchain-savers/components/ThorchainSaversManager/Deposit/components/Confirm.tsx

```diff
@@ -253,16 +286,119 @@ export const Confirm: React.FC<ConfirmProps> = ({ accountId, onNext }) => {
     ) as FeeDataEstimate<UtxoChainId> // We're lying to TS, this can be a FeeDataEstimate from any ChainId
   }, [getEstimateFeesArgs])
 
+  const getCustomTxInput: () => Promise<BuildCustomTxInput | undefined> = useCallback(async () => {
+    if (!contextDispatch) return
+    if (!(accountId && assetId && feeAsset && accountNumber !== undefined && wallet)) return
+    if (!state?.deposit.cryptoAmount) {
+      throw new Error('Cannot send 0-value THORCHain savers Tx')
+    }
+
+    try {
+      const chainAdapters = getChainAdapterManager()
+      const adapter = chainAdapters.get(chainId) as unknown as EvmChainAdapter
+
+      const amountCryptoBaseUnit = bnOrZero(state.deposit.cryptoAmount).times(
+        bn(10).pow(asset.precision),
```

Comment:
theres heaps of this pattern,  wonder if its worth adding a non-stringified `toBaseUnit` util

Reply:
Actually just using `toBaseUnit()` and be happy with the `toFixed(0)` version should achieve the same.
That's unless the base unit we get currently has decimals, which can definitely happen and would change the quote ever so slightly by an effectively non-noticeable margin, if this edge-case ever happens that is.

Do we want to go with `toBaseUnit()` for all these occurrences as carby syntax?

Reply:
Would be a nice feel-good change however I'll defer to you to decide based on your better understanding of the implications